x-pack/plugins/cases/server/client/attachments/bulk_delete.test.ts

```diff
@@ -9,9 +9,32 @@ import { loggerMock } from '@kbn/logging-mocks';
 import pReflect from 'p-reflect';
 import type { File } from '@kbn/files-plugin/common';
 import { FileNotFoundError } from '@kbn/files-plugin/server/file_service/errors';
-import { retrieveFilesIgnoringNotFound } from './bulk_delete';
+import { bulkDeleteFileAttachments, retrieveFilesIgnoringNotFound } from './bulk_delete';
+import { MAX_DELETE_FILES } from '../../../common/constants';
+import { createCasesClientMock, createCasesClientMockArgs } from '../mocks';
 
 describe('bulk_delete', () => {
+  describe('bulkDeleteFileAttachments', () => {
+    describe('errors', () => {
+      const casesClient = createCasesClientMock();
+      const clientArgs = createCasesClientMockArgs();
+
+      beforeEach(() => {
+        jest.clearAllMocks();
+      });
+
+      it(`throws 400 when trying to delete more than ${MAX_DELETE_FILES} files at a time`, async () => {
+        const fileIds = new Array(MAX_DELETE_FILES + 1).fill('fake-ids');
+
+        await expect(
+          bulkDeleteFileAttachments({ caseId: 'mock-id-4', fileIds }, clientArgs, casesClient)
+        ).rejects.toThrowError(
+          'Failed to delete file attachments for case: mock-id-4: Error: array must be of length <= 10'
```

Comment:
Same here about the message. It should be a bit more explicit.

Reply:
I'm not really sure if that can be done.

The schema/type has a message for when it fails but it has no knowledge of the variable name that is failing. So I cannot really say `FileIds error: array must be etc etc`.

One solution would be something like creating `limitedFileIdsArray` just so it could have a specific error message but it kind of defeats the purpose of having a generic `limitedArray` type.

Reply:
You can also pass a field name to schema and include in message. I am currently doing like this  ``The length of the ${field} is too long. The maximum length is ${max}.``

Reply:
oh even better.

Reply:
Updated the PR with this logic 🙌